### ISSUE TITLE
Initial Wallet Setup - Derivation Scheme UI

### ISSUE THREAD

**Issue by pavlenex** [btcpayserver/btcpayserver#1703]
## Background
When users register an account on their BTCPay Server, to use the software, they need to configure their wallet. Wallet configuration, is done in Store Settings > Derivation Scheme. This UX is far from ideal, but when BTCPay Server was first started, there was no wallet concept. To receive funds, users had to simply add their public key to the Store.

Today, BTCPay Server has several ways in which user scan configure their wallet. This setup is described in detail in [Wallet Setup Documentation.](https://docs.btcpayserver.org/WalletSetup/), to sum up, user can:

- Use an existing wallet
  - Existing hardware wallet
  - Existing software wallet
- Create a new wallet

## Video

I've made a video that guides you through this process and explains things. Hopefully a video is helpful.

https://youtu.be/5BHnr-VQU_c

## Wireframes
![New Mockup 1](https://user-images.githubusercontent.com/36959754/85927935-8d6df980-b8a9-11ea-9efe-7503641b6b9e.png)
![New Mockup 1 copy 9](https://user-images.githubusercontent.com/36959754/85927954-bc846b00-b8a9-11ea-9eed-a173f0172da2.png)
![New Mockup 1 copy 6](https://user-images.githubusercontent.com/36959754/85928012-26047980-b8aa-11ea-9ce2-aa368ce226d4.png)
![New Mockup 1 copy 2](https://user-images.githubusercontent.com/36959754/85928019-3157a500-b8aa-11ea-9f8b-4e33d59028aa.png)
![New Mockup 1 copy](https://user-images.githubusercontent.com/36959754/85927936-9068ea00-b8a9-11ea-834f-5014fc9a37cf.png)
![New Mockup 1 copy 5](https://user-images.githubusercontent.com/36959754/85928210-98c22480-b8ab-11ea-97cd-52acf1ef342c.png)
![New Mockup 1 copy 7](https://user-images.githubusercontent.com/36959754/85928219-a4155000-b8ab-11ea-8fd9-5aacca5b52db.png)
![New Mockup 1 copy 4](https://user-images.githubusercontent.com/36959754/85928222-a8da0400-b8ab-11ea-9d5e-6fbeac653d4a.png)
![New Mockup 1 copy 11](https://user-images.githubusercontent.com/36959754/85928226-ac6d8b00-b8ab-11ea-8a4b-a4311c974d03.png)

## Goal
The goal of this issue is to initiate the conversation on how we can improve this tricky UX. Hopefully it'll motivate experts to provide some tips, mockups or suggestions on what can be done. The mockups I provided are my own idea, so please feel free to provide your own. Issue should spark the convo between devs and UI/UX contributors and designers to come up with a solution that may not require huge refactoring but would simplify a very complicated workflow we now have in place. I encourage designers to test this process mainnet.demo.btcpayserver.org/ so they can get to know to it better.

**Additional context**

The UX can further be improved by using notifications we recently built into btcpay server, but lets leave that one for later.

If you need any clarification or have feedback on how to format this issue so it's better understanding for first time contributors, let me know. I'm @pavlenex on BTCPay Server Chat and Bitcoin Designers Slack.

Related issues: #1263 #1290

**Comment by raaans** [comment]
Who is the target demographic for this? Should it be super simplified for everyone to be able to do it or are we targeting hardcore bitcoiners? Do we want separate versions / flows so we can help both types of user and maybe a demographic in between?

**Comment by pavlenex** [comment]
Hey @raaans there's not the target demographic in BTCPay imo, since btcpay has quite a few [use-cases.](https://docs.btcpayserver.org/UseCase/) I'd say we should try to make this as simple as possible. I don't think we need separate versions for now, but if you think that's helpful we can try it out to see how it goes. 

How do you feel about current mockups? I am sure we can improve the UX quite a bit, looking forward to your ideas!

**Comment by pavlenex** [comment]
@raaans Fair point. I'd estimate our biggest user-group is merchants. Their technical level varies quite a bit, but let's say we want to make this as simple as possible for someone who has a small shop and wants to accept Bitcoin payments.

**Comment by pavlenex** [comment]
I've been pointed out to a really good [UX in Monero's wallet t](https://www.figma.com/file/tiU8sOzEZQq2kmsE89Vj21/Monero-Mobile-Concept?node-id=22%3A2422)hat has similar options. I think it's quite similar to what we're trying to set up here and can serve as a solid inspiration while designing our mockups.

**Comment by mrsellars** [comment]
I like the step by step approach a lot. I tried the current wallet set up by myself and even I'm technical skilled, it was not super easy and sometimes I had some question marks on my head. But thanks to your good documentation I had a connected wallet.

So maybe we can work in the wallet setup screens with a little bit more onboarding for the not so technical people. What do different terms mean and so on - could be in a first version just a link to different sections in the BTCPay Server documentation. In general when we design the pages for the two extremes:

- a) advanced user (very technical, good Bitcoin knowledge...) 
- b) newcomers (less technical, basic understanding of a wallet...) 

the screens should work also for regular users. It's a big challenge to design a workflow that works great for a) and b) but maybe we can handle it. Because you have already the existing screens, the migration path, how we get the current views adopted to a new layout would be also interesting. I can imagine that a breakdown into many steps would also create some serious development effort.

So long story short, what is a good place (maybe Figma?) to create some first draft flows and to get a conversation with the developers what is possible and what not?

**Comment by pavlenex** [comment]
Hey @mrsellars Yup, figma is great. You can join our community chat chat.btcpaysever.org on Mattermost to discuss any questions or ideas in real-time.  @dstrukt has a [Figma file](https://www.figma.com/file/C7Xyq0FlxgFW8vaBr8ht1z/BTCPAY?node-id=0%3A1) here that we use for this purpose.

**Comment by mrsellars** [comment]
@pavlenex cool! I will join later today and thanks for the link to Figma.

**Comment by Bosch-0** [comment]
Hey pavlenex, just an idea that popped into my head but not sure of the technical limitations. Is it possible to have a store setup process initiated upon signup followed by a wallet setup similar to your wire frames above **all before** the user lands on the initial BTCPay home screen? I think this alone would simplify things a lot. If this is not limited technically I'd be happy to do up some wire frames to illustrate what I mean in greater detail.

**Comment by pavlenex** [comment]
Hey @Bosch-0. In the long-run the idea is to get rid of that home page and replace it with dashboard #1711. We can explore the option of having some sort of wizard that would navigate user through this setup, but before that it would be good if we can figure out the ideal workflow that can happen with or without the wizard. Additionally, we can add better CTA on the dashboard itself for this. "Complete your account setup" and stuff like that.

It seems you and @mrsellars have some ideas for this, you can join our community chat.btcpayserver.org on Mattermost, so we can figure out best way to organize and collaborate on this.

**Comment by Bosch-0** [comment]
Fantastic idea. Is mattermost connected to your slack? Maybe we could create a #design channel on slack for people to discuss / share ideas pertaining to this and other design issues.

**Comment by pavlenex** [comment]
We do have a Slack bridge, but it's much better to join directly as for any serious discussion it gets weird with bots replying back and forth. We do have UX/UI channel on mattermost where we can begin discussion.

**Comment by dstrukt** [comment]
All W.I.P from tonight's session, but starting to form some basic structure

Derivation Scheme Flows
<img width="1655" alt="DerivationSchemeFlows" src="https://user-images.githubusercontent.com/6250771/89048843-3d011600-d305-11ea-9369-a8310df82606.png">

Derivation Scheme Detail
<img width="1194" alt="DerivationSchemeDetail" src="https://user-images.githubusercontent.com/6250771/89048849-3ecad980-d305-11ea-83ca-c2e966d65e13.png">

Derivation Scheme Flow B
![DerivationScheme](https://user-images.githubusercontent.com/6250771/89048856-40949d00-d305-11ea-983a-0e2cf5f60957.gif)

Rough mock of how the Derivation would degrade to mobile
<img width="443" alt="DerivationSchemeMobile" src="https://user-images.githubusercontent.com/6250771/89048868-42f6f700-d305-11ea-99dc-7e47e7763712.png">

All feedback is welcome!

**Comment by Eskyee** [comment]
very nice design, friendly & clear to follow, I think this is a big improvement for noobs using btcpay. wallet

**Comment by bolatovumar** [comment]
One question I have is about this screen. Are the top and bottom rows mutually exclusive and can not appear on the screen at once? If not, what's the difference between the top and bottom buttons?

![Screen Shot 2020-08-02 at 6 42 15 PM](https://user-images.githubusercontent.com/1934678/89138348-2a970000-d4f0-11ea-89cb-0fdcdb64269f.png)

**Comment by dstrukt** [comment]
@bolatovumar Oh :D they represent the same idea, just an exploration (with a visual element) on the same screen that wasn't cleaned up.

Making more progress on these flows, will update soon!

**Comment by dstrukt** [comment]
Making progress on this, should have something "finalized" by EOW, and we'll be reviewing this along with a few other tickets in Friday's design meeting (join us)!

Still thinking through a lot (particularly hardware flow), and how we might want to handle modifying / managing post-creation changes, but those will be issues regardless.

Few little prototype / wording issues I'm seeing when I recorded. If we went this path, the individual wording / edge-cases would probably be the area that would need the most work, but overall, you get the idea.

Point out flaws / dev considerations / opinions, etc.. so I can try to address them.

As usual, all feedback is welcome!

![WalletSetupV0 1](https://user-images.githubusercontent.com/6250771/95995567-7e726f00-0de6-11eb-86e7-95f20fecd88f.gif)

**Comment by dstrukt** [comment]
Further progress (posted a little while back on the Mattermost -- feel free to join and share your thoughts there as well:

![WalletSetupV0 2](https://user-images.githubusercontent.com/6250771/97116516-e4020d80-16ba-11eb-99d0-6436b849fcce.gif)

More updates to come, but getting closer

Additionally, working on before / after set-up states with @dennisreimann, as we improve the UX flow around stores and configuring up your wallet / lightning.

**Comment by Eskyee** [comment]
Looks very nice.. should we encourage a did you save your seed confirm.. ie with like three random words from seed... ??? Because if not double checking... the important thing of saving and seed backup is to be double sure at this stage... every thing else looks great.. and fluid

**Comment by dstrukt** [comment]
Ahh yeah, that's a good call .. I've always been curious if people find those helpful or cumbersome. I've always been incredibly particular and obsessive (for good reason), but wondered if others appreciate that degree of thoroughness.

I can definitely add that step and we can always cut it if people aren't feeling it!

**Comment by pavlenex** [comment]
[Notes from Design call #2](https://github.com/btcpayserver/organization/issues/16)

- Separate hot wallet vs watch-only wallet setup
- RPC wallet needs clarification ping Kukks to clarify and decide if we want this
- Account Settings remove when generating a new wallet
- Remove green separators
- Scan wallet QR code – add details about wallet support possibly in the scan tab as we do on the wallet file.
- Modify page when `enable` is clicked communicate it / Add new buttons 
- Enable/Disable on modify should be buttons

**Comment by pavlenex** [comment]
Done in #2164 :tada:
